crypto/rand/drbg_lib.c

```diff
@@ -229,6 +237,106 @@ int RAND_DRBG_reseed(RAND_DRBG *drbg,
 }
 
 /*
+ * Restart |drbg|, using the specified entropy or additional input
+ *
+ * Tries its best to get the drbg instantiated by all means,
+ * regardless of its current state.
+ *
+ * Optionally, a |buffer| of |len| random bytes can be passed,
+ * which is assumed to contain at least |entropy| bits of entropy.
+ *
+ * If |entropy| > 0, the buffer content is used as entropy input.
+ *
+ * If |entropy| == 0, the buffer content is used as additional input
+ *
+ * This function is used internally only.
+ */
+int rand_drbg_restart(RAND_DRBG *drbg,
+                      const unsigned char *buffer, size_t len, size_t entropy)
+{
+    int reseeded = 0;
+    const unsigned char *adin = NULL;
+    size_t adinlen = 0;
+
+    if (drbg->pool != NULL) {
+        RANDerr(RAND_F_RAND_DRBG_RESTART, ERR_R_INTERNAL_ERROR);
+        RAND_POOL_free(drbg->pool);
+        drbg->pool = NULL;
+        return 0;
```

Comment:
You could argue that it's not doing it's best (as the comment says to the function) to instantiate it. If you just call it again it will probably work. So I think it should continue or should not free the pool.

Reply:
It took me a while to get your point: It's not ok that the function fails on first call, but with a side effect that makes it (probably) succeed on the second call. So its more consistent to either succeed on the first call or fail ever after. I am going to choose the former (removing the `return 0`), in the hope that the if clause will never be entered anyway.